uiowa/uiowa
# Facilities API updates

Resolves #7401

# How to test
- Install composer updates.
```
 ddev composer install
```
- Add the following to your `docroot/sites/settings/local.settings.php` file:
```php
$config['uiowa_facilities.apis']['bizhub'] = [
  'auth' => [
    'user' => '',
    'pass' => '',
  ],
];
$config['uiowa_facilities.apis']['buildui'] = [
  'auth' => [
    'user' => '',
    'pass' => '',
  ],
];
```
- You can find the API credentials in ESAccess PW storage or ping me.
- While running commands, observe command line output and note if there are any messages that seem out of place or are missing.

## Buildings Import
```
ddev blt ds --site=facilities.uiowa.edu && ddev drush @facilities.local fm-buildings && ddev drush @facilities.local uli node/1731/edit
```

Observe no building import errors. Spot check building records look correct.

## Projects import
```
ddev blt ds --site=facilities.uiowa.edu && ddev drush @facilities.local fm-projects && ddev drush @facilities.local uli /all-projects
```
Observe no project import errors. Spot check project records look correct.

### Test graceful failure of basic auth
Our plan is to roll this out with the current prod URLs before basic auth has been added to them. To test this, adjust the config from above to add the current production endpoints and to remove the auth credentials, something like:
```php
$config['uiowa_facilities.apis']['bizhub'] = [
  'endpoint' => 'https://bizhub.facilities.uiowa.edu/bizhub/ext/',
//  'auth' => [
//    'user' => '',
//    'pass' => '',
//  ],
];
$config['uiowa_facilities.apis']['buildui'] = [
  'endpoint' => 'https://buildui.facilities.uiowa.edu/buildui/ext/',
//  'auth' => [
//    'user' => '',
//    'pass' => '',
//  ],
];
```
- Re-test both imports to ensure that they run without issue.

## Dispatch API
Re-test Dispatch API implementation. Use #4341 and #6269 as a reference.
